src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +144,1336 @@ static DamageTable D_80BBF858 = {
+                        return 0xDEC;
+                    }
+                    return 0xDEB;
+                }
+                this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+                return 0xDEA;
+            }
+            this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+            return 0xDE9;
+
+        case EN_S_GORO_TYPE_SHRINE_SPRING_D:
+            if (player->transformation == PLAYER_FORM_GORON) {
+                if (gSaveContext.save.weekEventReg[77] & 0x80) {
+                    if (this->textID == 0xDF1) {
+                        this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+                        return 0xDF2;
+                    }
+                    return 0xDF1;
+                }
+                this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+                return 0xDF0;
+            }
+            switch (this->textID) {
+                case 0xDED:
+                    return 0xDEE;
+                case 0xDEE:
+                    this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+                    return 0xDEF;
+                default:
+                    return 0xDED;
+            }
+            break;
+
+        case EN_S_GORO_TYPE_SHRINE_SPRING_E:
+            if (player->transformation == PLAYER_FORM_GORON) {
+                if (gSaveContext.save.weekEventReg[77] & 0x80) {
+                    if (this->textID == 0xDF6) {
+                        this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+                        return 0xDF7;
+                    }
+                    return 0xDF6;
+                }
+                this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+                return 0xDF5;
+            }
+            if (this->textID == 0xDF3) {
+                this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+                return 0xDF4;
+            }
+            return 0xDF3;
+
+        case EN_S_GORO_TYPE_SHRINE_SPRING_F:
+            if (player->transformation == PLAYER_FORM_GORON) {
+                if (gSaveContext.save.weekEventReg[77] & 0x80) {
+                    switch (this->textID) {
+                        case 0xDFB:
+                            return 0xDFC;
+                        case 0xDFC:
+                            this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+                            return 0xDFD;
+                        default:
+                            return 0xDFB;
+                    }
+                } else {
+                    this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+                    return 0xDFA;
+                }
+            } else {
+                if (this->textID == 0xDF8) {
+                    this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+                    return 0xDF9;
+                }
+                return 0xDF8;
+            }
+            break;
+
+        default:
+            return 0x0;
```

Comment:
```suggestion
            return 0;
```
For single digit numbers, we tend to not have `0x` prefix